sdk/tests/SerializationAndDeserializationTests.cc

```diff
@@ -0,0 +1,199 @@
+}
+
+TEST_F(SerializationAndDeserializationTests, DeserializeAccountIdFromProtobufTest)
+{
+  // Given
+  proto::AccountID testProtoAccountId = proto::AccountID();
+  testProtoAccountId.set_shardnum(123);
+  testProtoAccountId.set_realmnum(456);
+  testProtoAccountId.set_accountnum(789);
+
+  // When
+  AccountId accountId = AccountId::fromProtobuf(testProtoAccountId);
+  
+  // Then
+  EXPECT_EQ(accountId.toString(), "123.456.789");
+}
+
+TEST_F(SerializationAndDeserializationTests, SerializeContractIdТoProtobufTest)
+{
+  // Given
+  uint64_t testShardNum = getTestShardNum();
+  uint64_t testRealmNum = getTestRealmNum();
+  uint64_t testContractNum = getTestContractNum();
+  ContractId testContractId = ContractId(testShardNum, testRealmNum, testContractNum);
+
+  // When
+  auto protoContractId = std::unique_ptr<proto::ContractID>(testContractId.toProtobuf());
+
+  // Then
+  EXPECT_EQ(protoContractId->shardnum(), testShardNum);
+  EXPECT_EQ(protoContractId->realmnum(), testRealmNum);
+  EXPECT_EQ(protoContractId->contractnum(), testContractNum);
+}
+
+TEST_F(SerializationAndDeserializationTests, DeserializeContractIdFromProtobufTest)
+{
+  // Given
+  uint64_t testShardNum = getTestShardNum();
+  uint64_t testRealmNum = getTestRealmNum();
+  uint64_t testContractNum = getTestContractNum();
+  
+  proto::ContractID testProtoContractId = proto::ContractID();
+  testProtoContractId.set_shardnum(testShardNum);
+  testProtoContractId.set_realmnum(testRealmNum);
+  testProtoContractId.set_contractnum(testContractNum);
+
+  // When
+  ContractId contractId = ContractId::fromProtobuf(testProtoContractId);
+  
+  // Then
+  EXPECT_EQ(contractId.getShardNum(), testShardNum);
+  EXPECT_EQ(contractId.getRealmNum(), testRealmNum);
+  EXPECT_EQ(contractId.getContractNum(), testContractNum);
+}
+
+TEST_F(SerializationAndDeserializationTests, SerializeTransactionIdToProtobufTest)
+{
+  // Given
+  const std::string testAccountIdStr = "111.222.333";
+  const std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+  AccountId testAccountId(testAccountIdStr);
+  TransactionId testTransactionId = TransactionId::generate(testAccountId);
+
+  // When
+  const auto protoTransactionIdPtr = std::unique_ptr<proto::TransactionID>(testTransactionId.toProtobuf());
+  const auto protoTimestampPtr = std::unique_ptr<proto::Timestamp>(internal::TimestampConverter::toProtobuf(now));
+
+  // Then
+  EXPECT_EQ(static_cast<uint64_t>(protoTransactionIdPtr->accountid().shardnum()), testAccountId.getShardNum());
+  EXPECT_EQ(static_cast<uint64_t>(protoTransactionIdPtr->accountid().realmnum()), testAccountId.getRealmNum());
+  EXPECT_EQ(static_cast<uint64_t>(protoTransactionIdPtr->accountid().accountnum()), testAccountId.getAccountNum());
+  EXPECT_EQ(protoTransactionIdPtr->transactionvalidstart().seconds(), protoTimestampPtr->seconds());
+}
+
+TEST_F(SerializationAndDeserializationTests, DeserializeTransactionIdFromProtobufTest)
```

Comment:
To write this test I used test [ProtobufTransactionId](https://github.com/hashgraph/hedera-sdk-cpp/blob/c54139e8bc8f507e1f08d052b15eec3d188a0bd0/sdk/tests/TransactionIdTest.cc#L51) from [TransactionIdTest](https://github.com/hashgraph/hedera-sdk-cpp/blob/main/sdk/tests/TransactionIdTest.cc).

Reply:
Test `DeserializeTransactionIdFromProtobufTest` moved into `TransactionIdTest.cc` in commit https://github.com/hashgraph/hedera-sdk-cpp/pull/176/commits/672f803dcd2b2fc651dac7cd9f39e08cef329fff.